openfoam-interface/internal-interface/wrapper/simphonyInterface.C

```diff
@@ -945,6 +943,9 @@ void foam_modifyNumerics(std::string name, std::string fvSch, std::string fvSol,
         cDIS()
     );
 
+    TimeMod* t = (TimeMod*)(runTimes[name]);
```

Comment:
Ok.  I would then suggest that it would be clearer to do the following
1) Override 'virtual bool read()` in `TimeMod` (and include a comment there on why it differs from Time::`read())and therefore remove the `update`
2) Use `TimeMod` in `foam_init`:  ```Foam::Time *runTime = new TimeMod(controlDict_, "..", ".");```
3) Remove this cast at 946 and just call `runTimes[name].read()`